scripts/globals/abilities/climactic_flourish.lua

```diff
@@ -0,0 +1,54 @@
+-----------------------------------
+-- Ability: Ternary Flourish
+-- Description Allows you to deal critical hits. Requires at least one finishing move.
+-- Obtained: DNC Level 80
+-- Recast Time: 00:01:30 (Flourishes III)
+-- Duration: 00:01:00
+-- Cost: 1-5 Finishing Move charges
+-----------------------------------
+require("scripts/globals/settings")
+require("scripts/globals/status")
+require("scripts/globals/msg")
+-----------------------------------
+function onAbilityCheck(player, target, ability)
+
+    if (player:hasStatusEffect(tpz.effect.FINISHING_MOVE_1)) then
+        return 0, 0
+
+    elseif (player:hasStatusEffect(tpz.effect.FINISHING_MOVE_2)) then
+        return 0, 0
+
+    elseif (player:hasStatusEffect(tpz.effect.FINISHING_MOVE_3)) then
+        return 0, 0
+
+    elseif (player:hasStatusEffect(tpz.effect.FINISHING_MOVE_4)) then
+        return 0, 0
+
+    elseif (player:hasStatusEffect(tpz.effect.FINISHING_MOVE_5)) then
+        return 0, 0
+
+    else
+        return tpz.msg.basic.NO_FINISHINGMOVES, 0
+    end
+end
+
+function onUseAbility(player, target, ability)
+
+    if (player:hasStatusEffect(tpz.effect.FINISHING_MOVE_1)) then
+        player:delStatusEffect(tpz.effect.FINISHING_MOVE_1)
+        player:addStatusEffect(tpz.effect.CLIMACTIC_FLOURISH, 3, 0, 60, 0, player:getMerit(tpz.merit.CLIMACTIC_FLOURISH_EFFECT))
+    elseif (player:hasStatusEffect(tpz.effect.FINISHING_MOVE_2)) then
+        player:delStatusEffect(tpz.effect.FINISHING_MOVE_2)
+        player:addStatusEffect(tpz.effect.CLIMACTIC_FLOURISH, 3, 0, 60, 0, player:getMerit(tpz.merit.CLIMACTIC_FLOURISH_EFFECT))
+    elseif (player:hasStatusEffect(tpz.effect.FINISHING_MOVE_3)) then
+        player:delStatusEffect(tpz.effect.FINISHING_MOVE_3)
+        player:addStatusEffect(tpz.effect.CLIMACTIC_FLOURISH, 3, 0, 60, 0, player:getMerit(tpz.merit.CLIMACTIC_FLOURISH_EFFECT))
+    elseif (player:hasStatusEffect(tpz.effect.FINISHING_MOVE_4)) then
+        player:delStatusEffect(tpz.effect.FINISHING_MOVE_4)
+        player:addStatusEffect(tpz.effect.CLIMACTIC_FLOURISH, 3, 0, 60, 0, player:getMerit(tpz.merit.CLIMACTIC_FLOURISH_EFFECT))
+    elseif (player:hasStatusEffect(tpz.effect.FINISHING_MOVE_5)) then
+        player:delStatusEffect(tpz.effect.FINISHING_MOVE_5)
+        player:addStatusEffect(tpz.effect.CLIMACTIC_FLOURISH, 3, 0, 60, 0, player:getMerit(tpz.merit.CLIMACTIC_FLOURISH_EFFECT))
+    end
```

Comment:
I feel like this can be cleaned up, since:
```lua
    EFFECT_FINISHING_MOVE_1         = 381,
    EFFECT_FINISHING_MOVE_2         = 382,
    EFFECT_FINISHING_MOVE_3         = 383,
    EFFECT_FINISHING_MOVE_4         = 384,
    EFFECT_FINISHING_MOVE_5         = 385,
```
We can
```lua
for move = tpz.effect.FINISHING_MOVE_1, tpz.effect.FINISHING_MOVE_5 do
    player:delStatusEffect(move )
    player:addStatusEffect(tpz.effect.CLIMACTIC_FLOURISH, 3, 0, 60, 0, player:getMerit(tpz.merit.CLIMACTIC_FLOURISH_EFFECT))
end
```

Reply:
alright I set it up like that